core/templates/components/forms/custom-forms-directives/image-receiver.component.ts

```diff
@@ -39,36 +38,42 @@ interface ImageTypeMapping {
 }
 
 @Component({
-  selector: 'oppia-image-uploader',
-  templateUrl: './image-uploader.component.html',
+  selector: 'oppia-image-receiver',
+  templateUrl: './image-receiver.component.html',
 })
-export class ImageUploaderComponent {
+export class ImageReceiverComponent {
   @Output() fileChanged: EventEmitter<File> = new EventEmitter();
   // These properties are initialized using Angular lifecycle hooks
   // and we need to do non-null assertion. For more information, see
   // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
   @Input() allowedImageFormats!: string[];
-  @Input() isBlogPostThumbnailUploader!: boolean;
+  @Input() maxImageSizeInKB!: number;
   @ViewChild('dropArea') dropAreaRef!: ElementRef;
   @ViewChild('imageInput') imageInputRef!: ElementRef;
   fileInputClassName!: string;
   // The errorMessage will be null if the uploaded file is valid .
   errorMessage!: string | null;
   backgroundWhileUploading: boolean = false;
   licenseUrl = AppConstants.PAGES_REGISTERED_WITH_FRONTEND.LICENSE.ROUTE;
+  allowedImageFormatsString!: string;
+  maxAllowedImageSizeInBytes!: number;
```

Comment:
Usually the less state you have - the better.

Suppose you update maxImageSizeInKB at some point but forget to update this size in bytes, not good. This can be a private function

UPD: this applies to the field above as well.

Reply:
Yes, makes sense. We prefix private functions/members used within the class with an underscore so I created the _getMaxAllowedImageSizeInBytes() function.

And the functions which are used to generate strings for the frontend are not made private so just called the getAllowedImageFormatsString() directly.